eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -125,7 +125,8 @@ object Transactions {
   case class HtlcTimeoutTx(input: InputInfo, tx: Transaction, htlcId: Long) extends HtlcTx { override def desc: String = "htlc-timeout" }
   case class HtlcDelayedTx(input: InputInfo, tx: Transaction) extends TransactionWithInputInfo { override def desc: String = "htlc-delayed" }
   sealed trait ClaimHtlcTx extends TransactionWithInputInfo { def htlcId: Long }
-  case class ClaimHtlcSuccessTx(input: InputInfo, tx: Transaction, htlcId: Long) extends ClaimHtlcTx { override def desc: String = "claim-htlc-success" }
+  case class LegacyClaimHtlcSuccessTx(input: InputInfo, tx: Transaction, htlcId: Long) extends ClaimHtlcTx { override def desc: String = "claim-htlc-success" }
+  case class ClaimHtlcSuccessTx(input: InputInfo, tx: Transaction, paymentHash: ByteVector32, htlcId: Long) extends ClaimHtlcTx { override def desc: String = "claim-htlc-success" }
```

Comment:
I suppose it wasn't useful to define:

```scala
sealed trait ClaimHtlcSuccessTx extends ClaimHtlcTx
case class LegacyClaimHtlcSuccessTx extends ClaimHtlcSuccessTx 
case class ReplaceableClaimHtlcSuccessTx extends ClaimHtlcSuccessTx 
```

Reply:
No, it's actually not used anywhere for now. I'll see if it's necessary in the follow-up PR once I rebase it (#2102)